### PR TITLE
sys: migrate kubelet config to a file

### PR DESCRIPTION
**Pull request by george-angel**
https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/

### CODE REVIEW

File: worker.tf [hunk 1]
```diff
@@ -46,6 +44,25 @@ data "ignition_systemd_unit" "worker-kubelet" {
   content = "${data.template_file.worker-kubelet.rendered}"
 }
 
+data "template_file" "worker-kubelet-conf" {
+  template = "${file("${path.module}/resources/worker-kubelet-conf.yaml")}"
+
+  vars {
+    cluster_dns   = "${var.cluster_dns}"
+    feature_gates = "${join("\n  ", formatlist("%s: %s", keys(var.feature_gates), values(var.feature_gates)))}"
```
Comment (ffilippopoulos) [inline worker.tf]: we discussed yesterday about a comment for the spaces/yaml format that this one keeps, as that `"\n  ` is a bit odd

File: variables.tf [hunk 2]
```diff
@@ -182,5 +182,28 @@ variable "etcd_data_volumeids" {
...
+    "AdvancedAuditing"         = "false"
+    "ExpandPersistentVolumes"  = "true"
+    "PodShareProcessNamespace" = "true"
+  }
+}
+
+locals {
+  # Comma separated list for cli flas use, example output:
+  # `ExpandPersistentVolumes=true,PodShareProcessNamespace=true,AdvancedAuditing=false`
+  feature_gates_csv = "${join(",", formatlist("%s=%s", keys(var.feature_gates), values(var.feature_gates)))}"
```
Comment (ffilippopoulos) [inline variables.tf]: nice one!

File: resources/master-kubelet.service [hunk 3]
```diff
@@ -39,21 +39,16 @@ ExecStart=/usr/bin/docker \
...
     --node-labels=role=master,node-role.kubernetes.io/master="" \
     --register-with-taints=node-role.kubernetes.io/master=:NoSchedule \
-    ${feature_gates == "" ? "" : "--feature-gates=${feature_gates}"} \
     --container-runtime=docker \
     --network-plugin=cni \
     --cni-bin-dir=/opt/cni/bin \
     --cni-conf-dir=/etc/cni/net.d \
-    --allow-privileged \
-    --pod-manifest-path=/etc/kubernetes/manifests \
-    --minimum-container-ttl-duration=6m0s \
```
Comment (hectorhuertas) [inline resources/master-kubelet.service]: I don't see this flag replicated in the yaml config

File: resources/worker-kubelet.service [hunk 4]
```diff
@@ -38,24 +38,15 @@ ExecStart=/usr/bin/docker \
...
     --kubeconfig=/var/lib/kubelet/kubeconfig \
     --node-labels=role=${role} \
-    ${feature_gates == "" ? "" : "--feature-gates=${feature_gates}"} \
     --container-runtime=docker \
     --network-plugin=cni \
     --cni-bin-dir=/opt/cni/bin \
     --cni-conf-dir=/etc/cni/net.d \
-    --allow-privileged \
-    --pod-manifest-path=/etc/kubernetes/manifests \
-    --minimum-container-ttl-duration=6m0s \
```
Comment (hectorhuertas) [inline resources/worker-kubelet.service]: Idem

File: resources/master-kubelet.service (hunk 3 above)
Comment (george-angel) [reply resources/master-kubelet.service]: `Flag --minimum-container-ttl-duration has been deprecated, Use --eviction-hard or --eviction-soft instead. Will be removed in a future version.`

No, it was only put in recently coping OpenShift config, so I just removed it.

Comment (hectorhuertas) [reply resources/master-kubelet.service]: LGTM then!

**Pull request merged**